control/modelsimp.py

```diff
@@ -556,3 +557,216 @@ def markov(Y, U, m=None, transpose=False):
 
     # Return the first m Markov parameters
     return H if transpose else np.transpose(H)
+
+def okid(*args, m=None, transpose=False, dt=True, truncate=False):
+    """okid(Y, U, [, m])
+
+    Calculate the first `m` Markov parameters [D CB CAB ...]
+    from data.
+
+    This function computes the Markov parameters for a discrete time system
+
+    .. math::
+
+        x[k+1] &= A x[k] + B u[k] \\\\
+        y[k] &= C x[k] + D u[k]
+
+    given data for u and y.  The algorithm assumes that that C A^k B = 0 for
+    k > m-2 (see [1]_).  Note that the problem is ill-posed if the length of
+    the input data is less than the desired number of Markov parameters (a
+    warning message is generated in this case).
+
+    The function can be called with either 1, 2 or 3 arguments:
+
+    * ``H = okid(data)``
+    * ``H = okid(data, m)``
+    * ``H = okid(Y, U)``
+    * ``H = okid(Y, U, m)``
+
+    where `data` is an `TimeResponseData` object, and `Y`, `U`, are 1D or 2D
+    array and m is an integer.
+
+    Parameters
+    ----------
+    Y : array_like
+        Output data. If the array is 1D, the system is assumed to be
+        single input. If the array is 2D and transpose=False, the columns
+        of `Y` are taken as time points, otherwise the rows of `Y` are
+        taken as time points.
+    U : array_like
+        Input data, arranged in the same way as `Y`.
+    data : TimeResponseData
+        Response data from which the Markov parameters where estimated.
+        Input and output data must be 1D or 2D array.
+    m : int, optional
+        Number of Markov parameters to output. Defaults to len(U).
+    dt : True of float, optional
+        True indicates discrete time with unspecified sampling time and a
+        positive float is discrete time with the specified sampling time.
+        It can be used to scale the Markov parameters in order to match
+        the unit-area impulse response of python-control. Default is True
+        for array_like and dt=data.time[1]-data.time[0] for
+        TimeResponseData as input.
+    truncate : bool, optional
+        Do not use first m equation for least least squares. Default is False.
+    transpose : bool, optional
+        Assume that input data is transposed relative to the standard
+        :ref:`time-series-convention`. For TimeResponseData this parameter
+        is ignored. Default is False.
+
+    Returns
+    -------
+    H : ndarray
+        First m Markov parameters, [D CB CAB ...].
+
```

Comment:
Should we add a "See Also" section and reference `markov`?  Perhaps also a note here (and in `markov`) about how this differs from the `markov` command?

Reply:
Would that be enough?

See Also
--------
markov

Notes
-----
The :func:`~control.markov` command estimates the Markov parameters directly, which can be hard for slightly damped systems.
The :func:`~control.observer_kalman_identification` command uses a Kalman filter, which is better suited for slightly damped systems.